ethereum/core/src/main/java/org/hyperledger/besu/ethereum/chain/DefaultBlockchain.java

```diff
@@ -530,8 +555,15 @@ private void setGenesis(final Block genesisBlock) {
         throw new IllegalStateException("Blockchain is missing genesis block data.");
       }
       if (!genesisHash.get().equals(genesisBlock.getHash())) {
+        final String firstLine;
+        if (dataDirectory != null) {
+          firstLine =
+              "Supplied genesis block does not match stored chain data in " + dataDirectory + "\n";
+        } else {
+          firstLine = "Supplied genesis block does not match stored chain data.\n";
+        }
         throw new InvalidConfigurationException(
-            "Supplied genesis block does not match stored chain data.\n"
+            firstLine
                 + "Please specify a different data directory with --data-path or specify the original genesis file with --genesis-file.");
```

Comment:
I wonder if it's worth changing this to add something like "or --network if you're specifying a testnet or mainnet config"

Reply:
that makes sense! Good idea

Reply:
1368608